stages/org.osbuild.systemd.unit.create

```diff
@@ -0,0 +1,181 @@
+                            "simple",
+                            "exec",
+                            "forking",
+                            "oneshot",
+                            "dbus",
+                            "notify",
+                            "notify-reload",
+                            "idle"
+                        ]
+                    },
+                    "RemainAfterExit": {
+                        "type": "boolean"
+                    },
+                    "ExecStartPre": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    },
+                    "ExecStopPost": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    },
+                    "ExecStart": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    }
+                }
+            },
+            "Install": {
+                "additionalProperties": false,
+                "type": "object",
+                "description": "'Install' configuration section of a unit file.",
+                "properties": {
+                    "WantedBy": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    },
+                    "RequiredBy": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+"""
+
+
+def main(tree, options):
+    filename = options["filename"]
+    cfg = options["config"]
+    # ensure the service name does not exceed maximum filename length
+    if len(filename) > 255:
+        raise ValueError(f"Error: the {filename} service exceeds the maximum filename length.")
+
+    config = configparser.ConfigParser(allow_no_value=True)
+    # prevent conversion of the option name to lowercase
+    config.optionxform = lambda option: option
+
+    for section, opts in cfg.items():
+        if not config.has_section(section):
+            config.add_section(section)
+        for option, value in opts.items():
+            if isinstance(value, list):
+                for v in value:
+                    config.set(section, str(option) + "=" + str(v))
+            else:
+                config.set(section, option, str(value))
+
+    systemd_dir = f"{tree}/usr/lib/systemd/system"
+    with open(f"{systemd_dir}/{filename}", "w", encoding="utf8") as f:
+        config.write(f, space_around_delimiters=False)
+
+    ret = subprocess.run(["systemd-analyze", "verify", f"{systemd_dir}/{filename}"], check=True)
```

Comment:
The `check=True` and `return ret.returncode` is a bit redundant here as any non-zero return code will raise an error.